assets/js/util/index.js

```diff
@@ -115,73 +116,74 @@ export const removeURLParameter = ( url, parameter ) => {
-		number = number / 1000;
-		if ( 99 < number ) {
-			readableNumber = Math.round( number ) + 'K';
-		} else {
-			readableNumber = number.toFixed( 1 ) + 'K';
-		}
-	} else {
-		readableNumber = number;
-	}
+	// Cast parseable values to numeric types.
+	number = isFinite( number ) ? number : Number( number );
 
-	// Handle errors after calculations.
-	if ( isNull( number ) || isUndefined( number ) || isNaN( number ) ) {
-		readableNumber = '';
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
 		number = 0;
 	}
 
-	if ( 0 === number ) {
-		readableNumber = '0.00';
-		return currencyCode
-			? new Intl.NumberFormat( navigator.language, { style: 'currency', currency: currencyCode } ).format( number )
-			: number;
+	if ( currencyCode ) {
+		return numberFormat( number, { style: 'currency', currency: currencyCode } );
 	}
 
-	// Format as amount if currencyCode is passed.
-	if ( false !== currencyCode && '' !== readableNumber ) {
-		const formatedParts = new Intl.NumberFormat( navigator.language, { style: 'currency', currency: currencyCode } ).formatToParts( number );
-
-		const decimal = formatedParts.find( ( part ) => 'decimal' === part.type );
-		if ( ! isUndefined( decimal ) && ! isUndefined( decimal.value ) && 1000 > number ) {
-			readableNumber = Number.isInteger( number ) ? number : number.replace( '.', decimal.value );
-		}
-
-		const currencyFound = formatedParts.find( ( part ) => 'currency' === part.type );
-		const currency = currencyFound ? currencyFound.value : '';
+	const withSingleDecimal = {
+		minimumFractionDigits: 1,
+		maximumFractionDigits: 1,
+	};
 
-		return `${ currency }${ readableNumber }`;
+	switch ( true ) {
+		case 1000000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in millions.
+				__( '%sM', 'google-site-kit' ),
+				numberFormat( number / 1000000, withSingleDecimal )
+			);
+		case 99000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in thousands.
+				__( '%sK', 'google-site-kit' ),
+				numberFormat( Math.round( number / 1000 ) )
+			);
+		case 1000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in thousands.
+				__( '%sK', 'google-site-kit' ),
+				numberFormat( number / 1000, withSingleDecimal )
+			);
+		default:
+			return number.toString();
 	}
-
-	return readableNumber.toString();
 };
 
 /**
  * Internationalization Number Format.
  *
  * @param {number} number The number to format.
- * @param {string} locale Optional, locale to format as amount, default to Browser's locale.
+ * @param {Object} [options] Formatting options.
+ * @param {string} [locale] Locale tag. Optional.
  *
  * @return {string} The formatted number.
  */
-export const numberFormat = ( number, locale = '' ) => {
-	if ( ! locale ) {
-		locale = navigator.language;
-	}
+export const numberFormat = ( number, options = {}, locale = getLocale() ) => {
```

Comment:
I wonder whether `locale` should instead be `options.locale` - the current signature feels a bit strange to me, especially since the locale is another optional parameter that has a sane default.

Reply:
It's mostly for testing I think. I don't think you'd ever want/need to pass the locale directly, but I don't see why it shouldn't be supported. I agree the signature is a bit odd as-is.

We could implement it like this, which doesn't look bad

```js
export const numberFormat = ( number, options = {} ) => {
	const { locale = getLocale(), ...formatOptions } = options;

	return new Intl.NumberFormat( locale, formatOptions ).format( number );
};
```